app/services/idv/inherited_proofing/service_provider_forms.rb

```diff
@@ -0,0 +1,15 @@
+module Idv
+  module InheritedProofing
+    module ServiceProviderForms
+      include Idv::InheritedProofing::ServiceProviders
+
+      def inherited_proofing_form_for(service_provider, payload_hash:)
+        if service_provider == VA
+          return Idv::InheritedProofing::Va::Form.new payload_hash: payload_hash
+        end
+
+        raise 'Inherited proofing form could not be identified'
+      end
```

Comment:
I'm not seeing this method used at all outside of specs? Can we remove it, and then add it in the future PR that will actually use it? It appears very simple so I think it will not add too much bloat to that future PR

Reply:
update: ok I saw the sample usage in the PR description!

I think the mixin approach is counterproductive here (especially if we need two separate ones). If we moved to explicit, separate service and form classes we should be fine

```
class MyJob
  def perform
    form = VaForm.new(input_data)
    va_service = VaService.new
    va_service.check(input_data)
  end
end
```

Because at this point in the code, we know we'll be using the VA, so there's no need to try to detect multiple SPs?

Reply:
The mixins were split up that way on purpose because the _service_ is consumed by an `ActiveJob` job. Consuming the service results by the _form_ takes place once the job completes, in one of our flow steps (separate thread). Unless `DocumentCaptureSession` (redis) is able to store complex types (class that includes includes `ActiveModel::Model`), later to be retrieved by the flow step, it seems to make more sense to me to keep them separate. I have to confess, I am not that familiar with redis; is this possible without having to perform any serialization that will probably be more complex?

Reply:
redis can store anything, but we have tended to store basically structs as JSON. Take a look at this class: https://github.com/18F/identity-idp/blob/main/app/services/encrypted_redis_struct_storage.rb